src/consensus/encode.rs

```diff
@@ -679,6 +679,31 @@ impl Decodable for CheckedData {
     }
 }
 
+// References
+impl<'a, T: Encodable> Encodable for &'a T {
+    fn consensus_encode<S: io::Write>(&self, s: S) -> Result<usize, Error> {
+        (&**self).consensus_encode(s)
```

Comment:
I'm pretty sure the `&` isn't needed here and the method invocation magic will do the reference-taking. At least it still compiles for me on nightly and 1.29.0 without it.

Reply:
Yeah, it's not needed, but I think it's clearer.